Sources/Datadog/CrashReporting/CrashReporter.swift

```diff
@@ -20,18 +20,24 @@ internal class CrashReporter {
     // MARK: - Initialization
 
     convenience init?(
-        crashReportingFeature: CrashReportingFeature,
-        loggingFeature: LoggingFeature?,
-        rumFeature: RUMFeature?,
+        core: DatadogCoreProtocol,
         context: DatadogV1Context
     ) {
+        guard let feature = core.v1.feature(CrashReportingFeature.self) else {
+            return nil
+        }
+
         let loggingOrRUMIntegration: CrashReportingIntegration?
 
         // If RUM rum is enabled prefer it for sending crash reports, otherwise use Logging feature.
-        if let rumFeature = rumFeature {
-            loggingOrRUMIntegration = CrashReportingWithRUMIntegration(rumFeature: rumFeature, context: context)
-        } else if let loggingFeature = loggingFeature {
-            loggingOrRUMIntegration = CrashReportingWithLoggingIntegration(loggingFeature: loggingFeature, context: context)
+        if let rumFeature = core.v1.feature(RUMFeature.self) {
+            loggingOrRUMIntegration = CrashReportingWithRUMIntegration(
+                core: core,
+                rumConfiguration: rumFeature.configuration,
```

Comment:
This will be still preventing us from calling `DatadogCrashReporting` a standalone module, right? Do we have any thoughts on removing this `rumConfiguration` dependency?

For what I see, the real dependency in `CrashReportingWithRUMIntegration` is on:
* `rumConfiguration.sessionSampler`
* `rumConfiguration.backgroundEventTrackingEnabled`
* `rumConfiguration.uuidGenerator`
* `rumConfiguration.applicationID`

Reply:
Good catch, I can use parameters instead 👍 

We will still have feature coupling with the events types: `LogEvent`, `RUMErrorEvent`, and `RUMViewEvent` are still being shared.

Reply:
For events it's IMO fine - it's one category of code (and it's auto-generated). I'm wondering though how we will share these other things (except `applicationID` passed thru context, others don't have clear answer).